javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,260 @@
+    addFirst(e)
+    size() > sizeBefore
+  }
+
+  def offerLast(e: E): Boolean = {
+    val sizeBefore = size()
+    addLast(e)
+    size() > sizeBefore
+  }
+
+  def peekFirst(): E =
+    if (inner.isEmpty) null.asInstanceOf[E]
+    else inner.head.inner
+
+  def peekLast(): E =
+    if (inner.isEmpty) null.asInstanceOf[E]
+    else inner.last.inner
+
+  def pollFirst(): E = {
+    val ret = peekFirst()
+    if (ret == null) null.asInstanceOf[E] else {
+      inner.trimStart(1)
+      ret
+    }
+  }
+
+  def pollLast(): E = {
+    val ret = peekLast()
+    if (ret == null) null.asInstanceOf[E] else {
+      inner.trimEnd(1)
+      ret
+    }
+  }
+
+  def push(e: E): Unit =
+    addFirst(e)
+
+  def pop(): E =
+    removeFirst()
+
+  def removeFirstOccurrence(o: Any): Boolean = {
+    val index = indexOf(o)
+    if (index < 0 || index >= size()) false else {
+      remove(index)
+      true
+    }
+  }
+
+  def removeLastOccurrence(o: Any): Boolean = {
+    val index = lastIndexOf(o)
+    if (index < 0 || index >= size()) false else {
+      remove(index)
+      true
+    }
+  }
+
+  override def listIterator(_index: Int): ListIterator[E] = {
+    new ListIterator[E] {
+
+      if (_index < 0 || _index >= self.size())
+        throw new IndexOutOfBoundsException(_index.toString)
+
+      private var last: Int = _index
+      private var index: Int = _index
+
+      private var performedAdd = true
+      private var performedRemove = true
+
+      def hasNext(): Boolean =
+        (index < self.size && index >= 0)
+
+      def next(): E = {
+        if (!hasNext()) throw new NoSuchElementException
+        else {
+          performedAdd = false
+          performedRemove = false
+          index += 1
+          last = index - 1
+          self.inner(index - 1).inner
+        }
+      }
+
+      def hasPrevious(): Boolean =
+        (index > 0 && index <= self.size)
+
+      def previous(): E = {
+        if (!hasPrevious()) throw new NoSuchElementException
+        else {
+          performedAdd = false
+          performedRemove = false
+          index -= 1
+          last = index
+          self.inner(index).inner
+        }
+      }
+
+      def nextIndex(): Int =
+        if (index < self.size) self.size
+        else index + 1
+
+      def previousIndex(): Int =
+        if (index < 0) -1
+        else index - 1
+
+      def remove(): Unit = {
+        if (!performedAdd) {
+          self.remove(last)
+          index = last
+          performedRemove = true
+        } else throw new IllegalStateException()
+      }
+
+      def add(e: E): Unit = {
+        self.add(index, e)
+        index += 1
+        performedAdd = true
+      }
+
+      def set(e: E): Unit = {
+        if (!performedAdd && !performedRemove) self.set(index, e)
+        else throw new IllegalStateException()
+      }
+    }
+  }
+
+  def descendingIterator(): Iterator[E] = {
+    new Iterator[E] {
+      private val iter = inner.reverse.iterator
```

Comment:
You could use the `listIterator(self.size)` and use the `previous` on `next`, `hasPrevious` on `hasNext` and use the `remove` on it.